core/src/main/resources/org/kohsuke/stapler/bind.js

```diff
@@ -26,39 +26,43 @@ function makeStaplerProxy(url,crumb,methods) {
             for (var i=0; i<args.length-(callback!=null?1:0); i++)
                 a.push(args[i]);
 
-            if(window.jQuery === window.$) { //Is jQuery the active framework?
-                $.ajax({
-                    type: "POST",
-                    url: url+methodName,
-                    data: stringify(a),
-                    contentType: 'application/x-stapler-method-invocation;charset=UTF-8',
-                    headers: {'Crumb':crumb},
-                    dataType: "json",
-                    success: function(data, textStatus, jqXHR) {
-                        if (callback!=null) {
+            var headers = {
+                'Content-Type': 'application/x-stapler-method-invocation;charset=UTF-8',
+                'Crumb': staplerCrumb,
+            };
+            // If running in Jenkins, add Jenkins-Crumb header.
+            if (typeof crumb !== 'undefined') {
+                headers = crumb.wrap(headers);
+            }
```

Comment:
In the Prototype version, this was done deep inside of the guts of `Ajax.Request`, which Kohsuke had patched to automatically set this header. Now that we aren't using Jenkins' patched Prototype, we have to do this ourselves. This is a leaky abstraction, but it is not the only place in Stapler where we have Jenkins-specific logic.